homeassistant/components/swiss_public_transport/sensor.py

```diff
@@ -56,9 +56,9 @@ class SwissPublicTransportSensorEntityDescription(SensorEntityDescription):
     """Describes swiss public transport sensor entity."""
 
     exists_fn: Callable[[DataConnection], bool]
-    value_fn: Callable[[DataConnection], datetime | None]
+    value_fn: Callable[[DataConnection], Any]
 
-    index: int
+    index: int | None = None
```

Comment:
```suggestion
    index: int = 0
```
This way you can remove the `or 0` later on